backend/src/main/java/gov/cdc/usds/simplereport/config/FeatureFlagsConfig.java

```diff
@@ -16,28 +17,27 @@
 @EnableScheduling
 @Setter
 @Getter
+@RequiredArgsConstructor
+@Slf4j
 public class FeatureFlagsConfig {
-  @Autowired
```

Comment:
Is is @autowired and @RequiredArgsConstructor a replacement for each other?. I am trying to understand the things that I could have implemented better when I created this class 🤔

Reply:
So `Autowired` on a variable is [field injection](https://www.javaguides.net/2019/11/spring-field-injection-example-and-its-drawbacks.html), which means it uses magic to set the variable and makes it generally more difficult to test. 

Generally, the recommended way to do dependency injection is using constructor injection. In the case of a class having just one constructor spring will [implicitly add](https://www.baeldung.com/constructor-injection-in-spring#implicit-constructor-injection) the `@Autowired` on that constructor. You can read more about constructor vs field injection [here](https://medium.com/@toxicdev/field-injection-v-s-constructor-injection-dd9db2d85b7b).